internal/testutil/harnessutil/harnessutil.go

```diff
@@ -511,9 +515,97 @@ func (h *cachedCompilerHost) GetSourceFile(opts ast.SourceFileParseOptions) *ast
 	return result
 }
 
-func createCompilerHost(fs vfs.FS, defaultLibraryPath string, currentDirectory string) compiler.CompilerHost {
+type tracer struct {
+	fs                         vfs.FS
+	currentDirectory           string
+	packageJsonCache           map[tspath.Path]bool
+	builder                    strings.Builder
+	specifiTypesVersionMessage bool
+	noTypesVersionMessage      bool
+}
+
+func (t *tracer) trace(msg string) {
+	fmt.Fprintln(&t.builder, t.sanitizeTrace(msg))
+}
+
+func (t *tracer) sanitizeTrace(msg string) string {
+	if msg == "'package.json' has a 'typesVersions' field with version-specific path mappings." {
+		t.specifiTypesVersionMessage = true
+		t.noTypesVersionMessage = false
+		return msg
+	}
+	if msg == "'package.json' does not have a 'typesVersions' field." {
+		t.noTypesVersionMessage = true
+		t.specifiTypesVersionMessage = false
+		return msg
+	}
+	if strings.HasPrefix(msg, "'package.json' has a 'typesVersions' entry '") && !t.specifiTypesVersionMessage {
+		fmt.Fprintln(&t.builder, "'package.json' has a 'typesVersions' field with version-specific path mappings.")
+	}
+	if (msg == "'package.json' does not have a 'typings' field." ||
+		strings.HasPrefix(msg, "'package.json' has 'typings' field '")) &&
+		!t.noTypesVersionMessage {
+		fmt.Fprintln(&t.builder, "'package.json' does not have a 'typesVersions' field.")
+	}
+	t.noTypesVersionMessage = false
+	t.specifiTypesVersionMessage = false
+	// Version
+	if str := strings.Replace(msg, "'"+core.Version()+"'", "'"+FakeTSVersion+"'", 1); str != msg {
+		return str
+	}
+	// caching of fs in trace to be replaces with non caching version
```

Comment:
Just to be clear, this all is sort of a hack to work around the FS caching and produce consistent traces in tests?

Reply:
Yes..if you last few commits i didnt change anything in code just made traces consistent and show "caching" results in the traces or show them consistently when we can